gnucash/import-export/bi-import/dialog-bi-import.c

```diff
@@ -606,100 +605,94 @@ gnc_bi_import_create_bis (GtkListStore * store, QofBook * book,
         {
             g_string_assign(running_id, id);
             first_row_of_invoice = iter;
-        }
         
-        if (g_ascii_strcasecmp (type, "BILL") == 0)
-            invoice = gnc_search_bill_on_id (book, id);
-        else if (g_ascii_strcasecmp (type, "INVOICE") == 0)
-            invoice = gnc_search_invoice_on_id (book, id);
-        DEBUG( "Existing %s ID: %s\n", type, gncInvoiceGetID(invoice));
-
-        // If the search is empty then there is no existing invoice so make a new one
-        if (invoice == NULL)
-        {
-             DEBUG( "Creating a new : %s\n", type );
-            // new invoice
-            invoice = gncInvoiceCreate (book);
-            /* Protect against thrashing the DB and trying to write the invoice
-             * record prematurely */
-            gncInvoiceBeginEdit (invoice);
-            gncInvoiceSetID (invoice, id);
-            owner = gncOwnerNew ();
             if (g_ascii_strcasecmp (type, "BILL") == 0)
```

Comment:
This unwanted extra indentation makes this commit diff pretty hard to read... Can you fix it please ?

Reply:
I think the compare is confused because the same line of code appears twice in the source code; the second occurrence has an extra indent. The change here is fairly straightforward: previously, the code would check for an existing invoice for every row of the import; in the new version, the code does this check only for the first row of the invoice. So the code for 'check of invoice exists, and if not, create it' is now included in the 'if on_first_row_of_invoice' statement, so this code gets an extra indent. But the compare now matches second ocurrence of the said line of code in the old version, with the first occurrence to that line of code in the new version. That makes the compare difficult to read. I think the indents are ok here. I will try add make a comment, to prevent this 'compare confusion'.

Reply:
I can't remember what this was about exactly. In the final commit series I no longer see "unwanted indentation".